### PR TITLE
Bipartite graph spec

### PR DESCRIPTION
**Pull request by AnishShah**

### CODE REVIEW

**Comment by szymonm** [comment]
Hi @AnishShah, please fill your name on the list in the sheet here #91 whenever you start working on some test.

**Comment by bartekkalinka** [comment]
@AnishShah, I've put your name instead of mine in the BipartiteGraphSpec row of this [sheet]https://docs.google.com/spreadsheets/d/1x1Yc7DNsyS4FlOeOdLHlJEdZqL8QgU-TjLGbAOWG3wM/edit#gid=0

File: cassovary-core/src/test/scala/com/twitter/cassovary/graph/bipartite/BipartiteGraphSpec.scala [hunk 1]
```diff
@@ -13,19 +13,18 @@
...
-import com.twitter.cassovary.graph.GraphDir._
-import org.specs.Specification
+import org.scalatest.WordSpec
+import org.scalatest.matchers.ShouldMatchers
 import scala.collection.mutable
 
-class BipartiteGraphSpec extends Specification {
+class BipartiteGraphSpec extends WordSpec with ShouldMatchers {
 
   var graph: BipartiteGraph = _
```
Comment (pankajgupta) [inline cassovary-core/src/test/scala/com/twitter/cassovary/graph/bipartite/BipartiteGraphSpec.scala]: this is not your problem, but this test has been written previously pretty horribly (use of vars, side-effects, etc.) This is a good opportunity to fix it.

Please remove all vars and side-effects.

File: cassovary-core/src/test/scala/com/twitter/cassovary/graph/bipartite/BipartiteGraphSpec.scala [hunk 2]
```diff
@@ -155,95 +154,99 @@ class BipartiteGraphSpec extends Specification {
     graph = new BipartiteGraph(leftSide, rightSide, BipartiteGraphDir.Both)
   }
 
-  "Bipartite Graph Single Side Input" definedAs bipartiteExampleSingleSide should{
+  "Bipartite Graph Single Side Input" should {
     "have loaded correct number of nodes and edges" in {
```
Comment (pankajgupta) [inline cassovary-core/src/test/scala/com/twitter/cassovary/graph/bipartite/BipartiteGraphSpec.scala]: you can use:
val graph = bipartiteExampleSingleSide() before the 'in' statements and not have to change any of the 'in' tests

File: cassovary-core/src/test/scala/com/twitter/cassovary/graph/bipartite/BipartiteGraphSpec.scala [hunk 3]
```diff
@@ -13,19 +13,18 @@
...
+import org.scalatest.WordSpec
+import org.scalatest.matchers.ShouldMatchers
 import scala.collection.mutable
 
-class BipartiteGraphSpec extends Specification {
+class BipartiteGraphSpec extends WordSpec with ShouldMatchers {
 
   var graph: BipartiteGraph = _
   var leftNodes: Array[BipartiteNode] = _
   var rightNodes: Array[BipartiteNode] = _
```
Comment (pankajgupta) [inline cassovary-core/src/test/scala/com/twitter/cassovary/graph/bipartite/BipartiteGraphSpec.scala]: pls delete all vars

File: cassovary-core/src/test/scala/com/twitter/cassovary/graph/bipartite/BipartiteGraphSpec.scala [hunk 4]
```diff
@@ -90,7 +89,7 @@ class BipartiteGraphSpec extends Specification {
     graph = new BipartiteGraph(leftSide, rightSide, BipartiteGraphDir.LeftToRight)
```
Comment (pankajgupta) [inline cassovary-core/src/test/scala/com/twitter/cassovary/graph/bipartite/BipartiteGraphSpec.scala]: no need to set a graph variable. just have the last line be 
new BipartiteGraph(...

File: cassovary-core/src/test/scala/com/twitter/cassovary/graph/bipartite/BipartiteGraphSpec.scala [hunk 5]
```diff
@@ -155,95 +154,99 @@ class BipartiteGraphSpec extends Specification {
...
+      graph.getNodeById(0) shouldEqual None
+      graph.getNodeById(-10) shouldEqual None
+      graph.getNodeById(11) shouldEqual None
+      graph.getNodeById(150) shouldEqual None
     }
   }
 
-  "Bipartite Graph Double Sides input " definedAs bipartiteExampleDoubleSide should{
+  "Bipartite Graph Double Sides input " should {
     "have loaded correct number of nodes and edges" in {
```
Comment (pankajgupta) [inline cassovary-core/src/test/scala/com/twitter/cassovary/graph/bipartite/BipartiteGraphSpec.scala]: same as above, just have 
val graph = bipartiteExampleDoubleSide() here

File: cassovary-core/src/test/scala/com/twitter/cassovary/graph/bipartite/BipartiteGraphSpec.scala (hunk 3 above)
Comment (AnishShah) [inline cassovary-core/src/test/scala/com/twitter/cassovary/graph/bipartite/BipartiteGraphSpec.scala]: Should I remove all the vars? Even for inBounds, outBounds?

**Comment by AnishShah** [comment]
@pankajgupta I have made the changes. ```createLeftNode``` and ```createRightNode``` methods return the node and I'm assigning it to the array element. Is it OK? or is there anyway we can refactor this ?

**Comment by pankajgupta** [comment]
Your refactoring looks good

**Pull request merged**